google/cloud/storage/bucket.py

```diff
@@ -2785,6 +2799,49 @@ def object_retention_mode(self):
         if object_retention is not None:
             return object_retention.get("mode")
 
+    @property
+    def soft_delete_retention_duration_seconds(self):
```

Comment:
I see you're giving access to the soft delete policy's duration directly from the bucket, bypassing the actual soft delete policy object. 

Is this the right idea? I see how it's more convenient, but if in the future more fields get added to the policy, we would have to keep adding setters in the bucket

Reply:
We have both models in the python client - bucket property getter/setter for most of the metadata fields (e.g. retentionPolicy and Autoclass), or a dataclass representation for more complex fields (e.g. iamConfiguration and lifecycle). 

Do we expect more fields in the soft delete policy in the foreseeable future, specifically fields that will need to be exposed in the veneer client?

Skimming through phase 2 user guide, it doesn't seem like they are expanding `bucket.softDeletePolicy`. Instead I see separate bucket metadata fields `bucket.generation`, `bucket.hard_delete_time` etc

Reply:
Agreed that we don't want to overload the property getters/setters on the bucket. Updated to a data class following the bucket iamConfiguration pattern.
I don't think the original implementation would end up needed being deprecated; Though changing to a data class model will provide more flexibility if any changes come up in the next few phases.